src/Microsoft.ComponentDetection.Detectors/dockerfile/DockerfileComponentDetector.cs

```diff
@@ -36,19 +36,26 @@ public class DockerfileComponentDetector : FileComponentDetector, IDefaultOffCom
 
         protected override async Task OnFileFound(ProcessRequest processRequest, IDictionary<string, string> detectorArgs)
         {
-            var singleFileComponentRecorder = processRequest.SingleFileComponentRecorder;
-            var file = processRequest.ComponentStream;
-            var filePath = file.Location;
-            this.Logger.LogInfo($"Discovered dockerfile: {file.Location}");
+            try
+            {
+                var singleFileComponentRecorder = processRequest.SingleFileComponentRecorder;
+                var file = processRequest.ComponentStream;
+                var filePath = file.Location;
+                this.Logger.LogInfo($"Discovered dockerfile: {file.Location}");
+
+                string contents;
+                using (var reader = new StreamReader(file.Stream))
+                {
+                    contents = await reader.ReadToEndAsync();
+                }
 
-            string contents;
-            using (var reader = new StreamReader(file.Stream))
+                var stageNameMap = new Dictionary<string, string>();
+                var dockerFileComponent = this.ParseDockerFile(contents, file.Location, singleFileComponentRecorder, stageNameMap);
+            } catch (Exception e)
             {
-                contents = await reader.ReadToEndAsync();
+                this.Logger.LogError($"The file is not a Dockerfile it just has 'Docker' in the name. \n Error Message: <{e.Message}>");
+                this.Logger.LogException(e, isError: true, printException: true);
```

Comment:
I'm not sure we need the full exception printed here in quiet mode

```suggestion
                this.Logger.LogException(e);
```